packages/discord.js-utilities/src/lib/PaginatedMessages/PaginatedMessage.ts

```diff
@@ -831,9 +854,25 @@ export class PaginatedMessage {
 		}
 
 		if (this.response) {
-			await this.response.edit(page as WebhookEditMessageOptions);
+			if (runsOnInteraction(this.response)) {
+				if (this.response.replied || this.response.deferred) {
+					await this.response.editReply(page as WebhookEditMessageOptions);
+				} else {
+					await this.response.reply(page as WebhookEditMessageOptions);
+				}
+			} else {
+				await this.response.edit(page as WebhookEditMessageOptions);
+			}
+		} else if (runsOnInteraction(messageOrInteraction)) {
+			this.response = messageOrInteraction;
+
+			if (this.response.replied || this.response.deferred) {
+				await this.response.editReply(page as MessageOptions);
+			} else {
+				await this.response.reply(page as MessageOptions);
+			}
 		} else {
-			this.response = await channel.send(page as MessageOptions);
+			this.response = await messageOrInteraction.channel.send(page as MessageOptions);
```

Comment:
This should use followUp for interactions

Reply:
You put this comment on a line that deals with message based commands so it's unclear what exactly you're talking about.

Reply:
ohh, i see.